### PR TITLE
fix(accept): add space after every accepted word, not just the last one

### PR DESCRIPTION
**Pull request by FuJacob**
## Summary

"Add Space After Accepting" (#675) appeared to do nothing. The gate only fired when an accept **exhausted the whole suggestion** (`session.advancing(by: acceptedChunk.count).isExhausted`). Because the model almost always returns multi-word suggestions, accepting a single word — the normal case — never exhausts it, so no space was ever added unless you accepted the very last word or a one-word suggestion.

Now a word-ending accept also **consumes the suggestion's own following space**, so the trailing space lands with *this* accept on every word. This is done by extending the accepted chunk, not by synthesizing a character mid-suggestion, which is what keeps the delicate space system intact:

- Only whitespace the model already placed after the word is taken, so the accepted chunk stays an **exact prefix of the buffered suggestion** → the session stays in lockstep with the live field (consumed-suffix reconciliation still holds).
- The next chunk loses the leading space this one absorbed, and the reconciler's existing "drop leading whitespace when the field already ends in whitespace" rule means **no double space**.
- Punctuation (`world, how`), newlines (`line\nnext`), chunks already ending in punctuation (`done.`), and space-less scripts (CJK) are left untouched — the model's own layout wins there.
- The final-word / single-word case (no following space to consume) still appends via the existing exhaustion path, where the session tears down so a synthesized space can't desync anything.

Net: accept "world" from "world how are you" → inserts `world ` and the ghost continues with `how are you`; keep accepting and you get `world how are you `, or stop and type with the caret already spaced.

## Validation

```
xcodebuild ... build -derivedDataPath build/DerivedData                       # ** BUILD SUCCEEDED **
xcodebuild ... test -only-testing:CotabbyTests/SuggestionSessionReconcilerTests \
  -only-testing:CotabbyTests/SuggestionCoordinatorAcceptanceTests \
  CODE_SIGNING_ALLOWED=NO CODE_SIGNING_REQUIRED=NO                             # ** TEST SUCCEEDED ** (118 tests)
swiftlint lint --quiet                                                         # exit 0
```

Added 8 unit tests for the new `acceptanceChunkConsumingTrailingSpace` rule (following space, leading-whitespace chunks, whole horizontal runs, end-of-suggestion no-op, newline boundary, pre-punctuation no-op, chunk-ends-in-punctuation, space-less script).

## Risk / rollout notes

- Only affects behavior when the user has **"Add Space After Accepting" turned on** (still default off). With it off, `autoSpaceAdjustedChunk` returns the chunk unchanged — byte-for-byte prior behavior.
- The accepted chunk can now carry trailing whitespace; every consumer (insertion, session advance, word count, overlay caret prediction, `lastAcceptedTail`) handles that consistently because it is still a literal prefix of the suggestion tail.
- No settings/schema/persistence changes.

<h3>Greptile Summary</h3>

This PR fixes a bug where the "Add Space After Accepting" setting only fired when accepting the very last word of a suggestion (exhausting it), not on every mid-suggestion word accept. The fix moves the space-append logic from the insertion-text layer (synthesized, post-session) to the chunk layer (consuming the model's own following space as an exact prefix of the remaining text), so the session stays in lockstep with the live field on every accept.

- `autoSpaceAdjustedChunk` (new) extends the accepted chunk to include the suggestion's own trailing horizontal whitespace before the full insertion/session-advance/overlay pipeline runs, while `insertionTextApplyingAutoSpace` (existing, unchanged) still handles the exhausting-accept case where no following suggestion space exists to consume.
- 8 targeted unit tests for `acceptanceChunkConsumingTrailingSpace` cover following-space, multi-space runs, newline boundary, pre-punctuation, chunk-ends-in-punctuation, space-less script, and end-of-suggestion; two new coordinator-level integration tests verify the full path with the setting on/off.

<h3>Confidence Score: 5/5</h3>

Safe to merge — the change is gated behind the off-by-default 'add space after accepting' toggle, the new chunk-extension path is a pure function of data the session already holds, and the double-space interaction between the two spacing mechanisms is correctly precluded by existing guards.

The chunk-extension approach keeps the accepted chunk as an exact prefix of the session's remaining text, so every downstream consumer sees a consistent view. The exhaustion-time guard correctly no-ops when the extended chunk already ends in a space, preventing double-space. Both new coordinator-level integration tests and the 8 reconciler unit tests confirm the key boundary cases.

No files require special attention.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| Cotabby/Support/SuggestionSessionReconciler.swift | Adds `acceptanceChunkConsumingTrailingSpace`, a pure function that extends a word-ending accepted chunk by the suggestion's own horizontal whitespace. Guards are well-structured; newlines are excluded from the `prefix(while:)` predicate. |
| Cotabby/App/Coordinators/SuggestionCoordinator+Acceptance.swift | Wires `autoSpaceAdjustedChunk` into the acceptance path before the insertion chunk and auto-space exhaustion check. The existing `insertionTextApplyingAutoSpace` is unmodified and the double-space risk is precluded because `insertionChunkAppendingTrailingSpace` gates on a non-space last character. |
| CotabbyTests/SuggestionSessionReconcilerTests.swift | 8 new unit tests for `acceptanceChunkConsumingTrailingSpace` covering all documented boundary conditions. |
| CotabbyTests/SuggestionCoordinatorAcceptanceTests.swift | Adds two coordinator-level integration tests (setting ON and OFF) exercising the full pipeline. The `makeCoordinator` helper is updated to accept a `SuggestionSettingsSnapshot`. |

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant User
    participant Coordinator as SuggestionCoordinator
    participant Reconciler as SuggestionSessionReconciler
    participant Session as ActiveSuggestionSession
    participant Inserter as SuggestionInserter

    User->>Coordinator: acceptCurrentSuggestion()
    Coordinator->>Session: prepareAcceptance(granularity: .word)
    Session-->>Coordinator: .ready(liveContext, session, "world")

    Note over Coordinator: NEW: autoSpaceAdjustedChunk
    Coordinator->>Reconciler: acceptanceChunkConsumingTrailingSpace("world", remainingText:"world how")
    Reconciler-->>Coordinator: "world " (trailing space consumed)

    Coordinator->>Reconciler: insertionChunk(forAcceptedChunk:"world ", precedingText:"Hello")
    Reconciler-->>Coordinator: "world " (no leading-space strip needed)

    Coordinator->>Reconciler: insertionTextApplyingAutoSpace(insertionChunk:"world ", acceptedChunk:"world ", session)
    Note over Reconciler: session not exhausted, returns unchanged
    Reconciler-->>Coordinator: "world "

    Coordinator->>Inserter: insert("world ")
    Inserter-->>Coordinator: true

    Coordinator->>Session: commitAcceptedChunk("world ")
    Session-->>Coordinator: .advanced(remainingText:"how")

    Coordinator->>Coordinator: "state = .ready(text:"how")"
    Coordinator->>Coordinator: presentAdvancedOverlay(remainingText:"how")
```

<sub>Reviews (2): Last reviewed commit: ["test(accept): add coordinator-level inte..."](https://github.com/fujacob/cotabby/commit/76bcae26962ba4492dfd3ccbd51f40dcd876c2b9) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=37165756)</sub>

### CODE REVIEW

File: CotabbyTests/SuggestionSessionReconcilerTests.swift [hunk 1]
```diff
@@ -674,6 +674,66 @@ final class SuggestionSessionReconcilerTests: XCTestCase {
...
+        )
+    }
+
+    func test_acceptanceChunkConsumingTrailingSpace_skipsForSpacelessScript() {
+        // CJK scripts do not separate words with spaces, so even a stray following space is not taken.
+        XCTAssertEqual(
+            SuggestionSessionReconciler.acceptanceChunkConsumingTrailingSpace("資料", remainingText: "資料 です"),
+            "資料"
+        )
+    }
```
[the remark this answers is not in the document]
Comment (FuJacob) [reply CotabbyTests/SuggestionSessionReconcilerTests.swift:735]: Good call — added two coordinator-level integration tests in 76bcae2 (`test_acceptCurrentSuggestion_withAddSpaceAfterAccept_insertsTrailingSpaceOnNonFinalWord` and the OFF counterpart). They drive the full path through acceptCurrentSuggestion() with a multi-word suggestion and assert the inserted chunk and remaining tail, so a coordinator-layer wiring break is caught.
